amplify/function/BusinessLogic/ResetHackathon/handler.ts

```diff
@@ -216,77 +199,121 @@ export const handler: Handler = async (event) => {
       }
     }
 
-    // Edit start date
-    if (startDate) {
-      console.log("editing start date");
-      const { errors } = await client.graphql({
-        query: updateHackathon,
-        variables: {
-          input: {
-            id: HackathonID,
-            startDate: startDate,
+    // get the current hackathon model data items
+    const { data: hackathonItems } = await client.graphql({
+      query: listHackathons,
+    });
+    const HackathonItems = hackathonItems.listHackathons.items;
+
+    if (!(HackathonItems.length === 0)) {
```

Comment:
Is it possible to use this check solely to run either create or update  and then do rest of the logic such as start date/end date/scoring components once? Right now the logic is duplicated between creating and updating even though it is mostly similar until the part where we actually update/create. All the separate update calls could be consolidated into one, I believe.